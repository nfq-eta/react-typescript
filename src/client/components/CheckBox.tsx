import * as React from 'react';
import * as styles from './CheckBox.css';

export interface IItem {
    value: string | number;
    label: string;
}

export interface IProps {
    item: IItem;
    handleClick: (item: IItem, checked: boolean) => void;
    checked?: boolean;
}

interface ICheckBoxState {
    checked: boolean;
}

export default class CheckBox extends React.Component<IProps, ICheckBoxState> {
    constructor(props: IProps) {
        super();

        this.state = {
            checked: props.checked || false,
        };
    }

    handleClick = () => {
        const checked = !this.state.checked;
        this.setState({ checked });

        const { value, label } = this.props.item;
        this.props.handleClick({ value, label }, checked);
    };

    render() {
        const { value, label } = this.props.item;

        return (
            <div
                key={'a' + value}
                onClick={this.handleClick}
                className={this.state.checked ? styles.checkBox : styles.checkBoxSelected}
            >
                <label>
                    <input type="checkbox" value={value} checked={this.state.checked} />
                    {label}
                </label>
            </div>
        );
    }
}
