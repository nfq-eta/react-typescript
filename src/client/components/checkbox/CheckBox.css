.checkBox {
    cursor: pointer;
}

.checkBoxSelected {
    composes: checkBox;
    color: #ff1323;
}
