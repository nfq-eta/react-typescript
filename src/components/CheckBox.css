.checkBox {
    cursor: pointer;
}
.checkBoxSelected {
    composes: checkBox;
    color: red;
}
