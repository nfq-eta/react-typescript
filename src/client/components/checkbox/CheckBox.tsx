import * as React from 'react';
import * as styles from './CheckBox.css';

export interface IItem {
    id: string;
    value: string | number;
    label: string;
}

export interface IProps {
    item: IItem;
    handleClick: (item: IItem, checked: boolean) => void;
    checked?: boolean;
}

interface ICheckBoxState {
    checked: boolean;
}

export interface ITestEvent {
    target: { checked: boolean };
}

export default class CheckBox extends React.Component<IProps, ICheckBoxState> {
    constructor(props: IProps) {
        super(props);

        this.state = {
            checked: props.checked || false,
        };
    }

    handleClick = (e: React.ChangeEvent<HTMLInputElement> | ITestEvent) => {
        const checked = e.target.checked;
        this.setState({ checked });

        this.props.handleClick(this.props.item, checked);
    };

    render() {
        const { value, label } = this.props.item;

        return (
            <label className={this.state.checked ? styles.checkBox : styles.checkBoxSelected}>
                <input
                    type="checkbox"
                    onChange={this.handleClick}
                    value={value}
                    defaultChecked={this.state.checked}
                />
                {label}
            </label>
        );
    }
}
