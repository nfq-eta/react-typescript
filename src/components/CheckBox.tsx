import * as React from 'react';
import * as styles from './CheckBox.css';

export interface IItem {
    id: number;
    label: string;
}

export interface IProps {
    item: IItem;
    handleClick: (item: IItem, checked: boolean) => void;
    checked?: boolean;
}

interface ICheckBoxState {
    checked: boolean;
}

export default class CheckBox extends React.Component<IProps, ICheckBoxState> {
    constructor(props: IProps) {
        super();

        this.state = {
            checked: props.checked || false,
        };
    }

    handleClick = () => {
        const checked = !this.state.checked;
        this.setState({ checked });

        const { id, label } = this.props.item;
        this.props.handleClick({ id, label }, checked);
    };

    render() {
        const { id, label } = this.props.item;

        return (
            <div
                key={id}
                onClick={this.handleClick}
                className={this.state.checked ? styles.checkBox : styles.checkBoxSelected}
            >
                {id}. {!this.state.checked ? '[ ] ' : '[x] '}{label}
            </div>
        );
    }
}
