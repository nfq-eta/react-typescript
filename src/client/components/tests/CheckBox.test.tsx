import * as React from 'react';
import { shallow } from 'enzyme';
import enzymeToJson from 'enzyme-to-json';
import {} from 'jest';

import CheckBox from '../CheckBox';

describe('CheckBox.js', () => {
    const props = {
        item: {
            value: 1,
            label: 'Test',
        },
        handleClick: jest.fn(),
    };

    it('renders without crashing', () => {
        shallow(<CheckBox {...props} />);
    });

    it('matches snapshot', () => {
        const wrapper = shallow(<CheckBox {...props} />);
        expect(enzymeToJson(wrapper)).toMatchSnapshot();
    });

    it('Label should be visible', () => {
        const wrapper = shallow(<CheckBox {...props} />);
        expect(wrapper.find('div').text()).toContain('Test');
    });

    it('should be checked', () => {
        const wrapper = shallow(<CheckBox {...props} checked={true} />);
        const appInstance = wrapper.instance() as CheckBox;
        expect(appInstance.state.checked).toBeTruthy();
        expect(enzymeToJson(wrapper)).toMatchSnapshot();
    });

    it('should toggle state', () => {
        const wrapper = shallow(<CheckBox {...props} />);
        const appInstance = wrapper.instance() as CheckBox;
        appInstance.handleClick();

        expect(appInstance.state.checked).toBeTruthy();

        appInstance.handleClick();
        expect(appInstance.state.checked).toBeFalsy();
    });
});
