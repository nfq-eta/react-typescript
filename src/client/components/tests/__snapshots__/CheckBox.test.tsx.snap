// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckBox.js matches snapshot 1`] = `
<div
  className="checkBoxSelected"
  key="a1"
  onClick={[Function]}
>
  <label>
    <input
      checked={false}
      type="checkbox"
      value={1}
    />
    Test
  </label>
</div>
`;

exports[`CheckBox.js should be checked 1`] = `
<div
  className="checkBox"
  key="a1"
  onClick={[Function]}
>
  <label>
    <input
      checked={true}
      type="checkbox"
      value={1}
    />
    Test
  </label>
</div>
`;
